### ISSUE TITLE
Pressing Regenerate Invoices button does not regenerate the invoice

### ISSUE THREAD

**Issue by sewcialite** [AccelerateNetworks/NumberSearch#344]
**Describe the bug**
The regenerate invoice button does not update the quote and gives out bad information, 

**To Reproduce**
1. Go to ops
2. Click on regenerate invoices
3. Top Bar says "Successfully deleted the existing invoices and created new Invoices for this quote"
4. It does not update the quote and the Top Bar should say, "Quote has been updated" because we no longer delete the previous invoice. 

![image](https://github.com/AccelerateNetworks/NumberSearch/assets/87674023/c1cef583-6164-4254-b2e9-86e5b7705e2c)

**Comment by uncheckederror** [comment]
@sewcialite Unfortunately you haven't really given me any information in this bug report. What I need to know to investigate this issue is a link to the Order that broke in the Ops app (copy and paste the URL) and a description of why you were trying to regenerate the invoices for that specific order. Without those two pieces of information there's nothing I can do to solve this problem.

**Comment by danry25** [comment]
@uncheckederror Press regenerate on this quote: https://ops.acceleratenetworks.com/Home/Order/58b7dd38-0073-4cee-82d3-e09c4d337bbf

**Comment by uncheckederror** [comment]
> @uncheckederror Press regenerate on this quote: https://ops.acceleratenetworks.com/Home/Order/58b7dd38-0073-4cee-82d3-e09c4d337bbf

Thanks for the link!

**Comment by uncheckederror** [comment]
This order has an Oregon zip code "97301" so the tax rate lookup is failing. Because a valid tax rate cannon be found and invalid tax rate is getting passed to the billing system with a name of ", WA - -1" and a value of "-100M".

@danry25 when we can't find a tax rate for an order because it's not in WA what tax rate name and value should be set it based on the options on this page: https://ops.acceleratenetworks.com/Home/TaxRates

**Comment by danry25** [comment]
@uncheckederror Well, kQBeX5layK tax rate should not exist, we can never have negative tax rates. Lets not have Ops set a tax rate at all on orders outside Washington State.

**Comment by uncheckederror** [comment]
Okay, so that issue is resolved. This process is now failing because the id of the invoice saved to the order in Ops system results in a 404 when we query for it in the billing system. Did someone delete or recreate the original one-time invoice? How is the Ops system supposed to handle this.

**Comment by uncheckederror** [comment]
Okay, there were a bunch of different rabbit holes I went down here but the core issue was that invoices and quotes are queried separately in invoiceninja and for this Quote we were trying to find matching invoices which didn't exist. This issue has been resolved, if you hit the generate invoices button in the Ops app it will work now.

I have also made major improvements to the UX when failures to interact with InvoiceNinja occur.
